packages/core/src/command/hl7v2-subscriptions/hl7v2-to-fhir-converter.ts

```diff
@@ -0,0 +1,324 @@
+import { Hl7Message } from "@medplum/core";
+import { Bundle, Resource } from "@medplum/fhirtypes";
+import { errorToString, executeWithNetworkRetries } from "@metriport/shared";
+import axios from "axios";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { S3Utils, StoreInS3Params, storeInS3WithRetries } from "../../external/aws/s3";
+import { deduplicate } from "../../external/fhir/consolidated/deduplicate";
+import { toFHIR as toFhirPatient } from "../../external/fhir/patient/conversion";
+import { buildBundle, buildBundleEntry } from "../../external/fhir/shared/bundle";
+import { capture, out } from "../../util";
+import { Config } from "../../util/config";
+import { JSON_APP_MIME_TYPE } from "../../util/mime";
+import { convertHl7v2MessageToFhir } from "./hl7v2-to-fhir-conversion";
+import { Hl7MessageType, getHl7MessageTypeOrFail } from "./hl7v2-to-fhir-conversion/msh";
+import {
+  buildHl7MessageCombinedBundleFileKey,
+  buildHl7MessageFhirBundleFileKey,
+} from "./hl7v2-to-fhir-conversion/shared";
+
+dayjs.extend(duration);
+
+const region = Config.getAWSRegion();
+
+// TODO 2887: Add more event types to both arrays
+const createMessageTypes = ["A01"];
+const updateMessageTypes = ["A03"];
+
+const INTERNAL_HL7_ENDPOINT = `internal/hl7`;
+const INTERNAL_GET_PATIENT_ENDPOINT = "internal/patient";
+const SIGNED_URL_DURATION_SECONDS = dayjs.duration({ minutes: 10 }).asSeconds();
+
+type Hl7ToFhirLambdaProps = {
+  cxId: string;
+  patientId: string;
+  message: string;
+  messageId: string;
+  messageReceivedTimestamp: string;
+  apiUrl: string;
+  bucketName: string;
+};
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+
+export async function convertHl7MessageToFhirAndUpload({
+  cxId,
+  patientId,
+  message,
+  messageId,
+  messageReceivedTimestamp,
+  apiUrl,
+  bucketName,
+}: Hl7ToFhirLambdaProps): Promise<void> {
+  const { log } = out(`Hl7 to FHIR Lambda - cx: ${cxId}, pt: ${patientId}`);
+  log(`Running message from ${messageReceivedTimestamp}`);
+  const hl7Message = Hl7Message.parse(message);
+  const s3Client = getS3UtilsInstance();
+  const internalHl7RouteUrl = `${apiUrl}/${INTERNAL_HL7_ENDPOINT}`;
+  const internalGetPatientUrl = `${apiUrl}/${INTERNAL_GET_PATIENT_ENDPOINT}/${patientId}?cxId=${cxId}`;
+
+  const msgType = getHl7MessageTypeOrFail(hl7Message);
+  const shouldNotifyApi =
+    createMessageTypes.includes(msgType.triggerEvent) ||
+    updateMessageTypes.includes(msgType.triggerEvent);
```

Comment:
noting that there may be update message type - like maybe an A08 or something - where we do not want to notify the api, fine for now, but noting since we dont want to accidentally start sending webhooks for all sorts of events.